plugins/bundle/config.go

```diff
@@ -49,6 +52,13 @@ func ParseConfig(config []byte, services []string) (*Config, error) {
 // the defined `bundles`. This expects a map of bundle names to resource
 // configurations.
 func ParseBundlesConfig(config []byte, services []string) (*Config, error) {
+	return ParseBundlesConfigWithSigning(config, services, nil)
+}
+
+// ParseBundlesConfigWithSigning validates the config and injects default values for
+// the defined `bundles`. This expects a map of bundle names to resource
+// configurations and public keys to verify a signed bundle.
+func ParseBundlesConfigWithSigning(config []byte, services []string, keys map[string]*bundle.KeyConfig) (*Config, error) {
```

Comment:
One of the nice things about using the builder pattern is that you don't have to add extra variants every time the API needs to be extended. For example:

```
type ConfigBuilder struct {
  raw []byte
  services []string
  keys map[string]*bundle.KeyConfig
}

func NewConfigBuilder() *ConfigBuilder { ... }
func (b *ConfigBuilder) WithRawBytes([]byte) *ConfigBuilder {...}
func (b *ConfigBuilder) WithServices([]string) *ConfigBuilder {...}
func (b *ConfigBuilder) WithKeyConfigs(map[string]*bundle.KeyConfig) {...}
func (b *ConfigBuilder) Parse() (*Config, error)
```

Reply:
Updated the bundle and discovery config code to use builder pattern.